src/objective/objective_function.cpp

```diff
@@ -43,6 +47,15 @@ ObjectiveFunction* ObjectiveFunction::CreateObjectiveFunction(const std::string&
   } else if (type == std::string("constrained_cross_entropy")) {      // Entry-point for FairGBM code!
     return new ConstrainedCrossEntropy(config);
   } else if (type == std::string("constrained_recall_objective")) {   // Entry-point for FairGBM code!
+    if(config.objective_stepwise_proxy == "quadratic") {
+      return new ConstrainedRecallObjectiveQuadratic(config);
```

Comment:
the hierarchy is not supposed to be under the ObjectiveClass; it's a new hierarchy of proxy losses, and those new classes from the hierarchy will be used in both constrained xentropy and constrained recall objectives.